### DISCUSSION TITLE
Webpack `config.optimization.minimize = true` should be the default

### DISCUSSION THREAD

**Discussion by nicolassanmar** [vercel/next.js#51888] [Ideas]
### Goals

I think we should minify our server code by default (on `next build`).  
Currently, the way to do this is to add a webpack config option, which is somewhat out of the way.
To add this, we are required to add:

``` ts
    webpack: (config) => {
      if (config.name === 'server') {
        config.optimization.minimize = true
        config.devtool = 'hidden-source-map'
      }
      return config;
    },

```

### Background

I think this is a reasonable default to both:
1. Decrease server bundle size, which should decrease cold start times on serverless functions 
2. Make it easily compatible with other tools. For example, I've had some issues using `Sentry` as server side code was not minified.

I would like to know on which scenarios people actually want their server code not to be minified. After considering the use-cases and the amount of people that prefer one thing over the other, we can figure out if this proposal makes more sense as a default or as a more direct config option.

### Proposal

As stated, I think we should minimized code the default, and make it so that users can opt out of it using `webpack` config, as I think that case is less common.

If that's not something we want to do, we could at least provide a simple config option on the top-level of the `module.exports` on `next.config.js`
A way to implement this could be the following
```
minifyServer: true // defaults to false, maps to `config.optimization.minimize = true`
minifyServerDevtool: 'hidden-source-map' // maps to `config.devtool`
```

I am willing to contribute if this proposal seems reasonable

**Comment by CogitoBurrito** [comment]
https://github.com/vercel/next.js/pull/51559

**Reply by nicolassanmar** [reply]
That PR just got closed and not merged

**Comment by giulianovarriale** [comment]
@nicolassanmar One reason to avoid minifying server-side code is to preserve clear stack traces in the logs. I'm currently exploring options to improve this using source maps, but Datadog doesn’t support it for logs ingested from the server side. From what I’ve found, enabling source maps at runtime impacts performance. So, the simplest and cheapest solution might be to just avoid minifying the source code.
